src/rgw/rgw_rados.cc

```diff
@@ -8354,8 +8354,12 @@ int RGWRados::delete_system_obj(rgw_obj& obj, RGWObjVersionTracker *objv_tracker
   return 0;
 }
 
-int RGWRados::delete_obj_index(rgw_obj& obj)
+int RGWRados::delete_obj_index(rgw_obj& _obj)
 {
+  rgw_obj obj = _obj;
+  if (obj.get_instance() == "null") {
```

Comment:
@yehudasa In delete_obj, before we operate on index ,we clear instance if it's "null".
`int RGWRados::Object::Delete::delete_obj()
{
  RGWRados *store = target->get_store();
  rgw_obj& src_obj = target->get_obj();
  const string& instance = src_obj.get_instance();
  rgw_obj obj = src_obj;

  if (instance == "null") {
    obj.clear_instance();
  }`
So in delete_obj_index, I think we should also do this. I test this and it works.